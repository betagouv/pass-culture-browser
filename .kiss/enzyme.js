import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
