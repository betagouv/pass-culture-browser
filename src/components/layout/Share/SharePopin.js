import { connect } from 'react-redux'
import { compose } from 'redux'

import SharePopinContent from './SharePopinContent'

const mapStateToProps = ({ share }) => ({ ...share })

export const SharePopin = compose(connect(mapStateToProps))(SharePopinContent)

export default SharePopin
