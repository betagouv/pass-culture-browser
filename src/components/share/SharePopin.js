import { compose } from 'redux'
import { connect } from 'react-redux'

import SharePopinContent from './SharePopinContent'

const mapStateToProps = ({ user, share }) => {
  const email = (user && user.email) || ''
  return { email, ...share }
}

export const SharePopin = compose(connect(mapStateToProps))(SharePopinContent)

export default SharePopin
