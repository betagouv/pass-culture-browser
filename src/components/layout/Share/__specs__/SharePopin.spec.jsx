import React from 'react'
import { shallow } from 'enzyme'

import SharePopin from '../SharePopin'

const dispatchMock = jest.fn()

describe('src | components | share | SharePopin', () => {
  it('should match the snapshot', () => {
    // given
    const props = {
      dispatch: dispatchMock,
      options: {
        text: 'Fake Test',
        title: 'Fake Title',
        url: 'fake@url.com',
      },
      visible: true,
    }

    // when
    const wrapper = shallow(<SharePopin {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('functions', () => {
    describe('closeHandler', () => {
      describe('when options are true', () => {
        it('should call dispatch with good action parameters', () => {
          // given
          const props = {
            dispatch: dispatchMock,
            email: 'fake@email.com',
            options: {
              text: 'Fake Test',
              title: 'Fake Title',
              url: 'fake@url.com',
            },
            visible: true,
          }

          // when
          const wrapper = shallow(<SharePopin {...props} />).dive()
          wrapper.find('button').simulate('click')
          const expected = {
            options: null,
            type: 'TOGGLE_SHARE_POPIN',
          }

          // then
          expect(dispatchMock).toHaveBeenCalledWith(expected)
        })
      })
    })
  })
})
