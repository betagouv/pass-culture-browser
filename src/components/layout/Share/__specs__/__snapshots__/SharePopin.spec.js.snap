// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | share | SharePopin should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SharePopin
    dispatch={[MockFunction]}
    options={
      Object {
        "text": "Fake Test",
        "title": "Fake Title",
        "url": "fake@url.com",
      }
    }
    visible={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appear": false,
      "children": [Function],
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "timeout": 500,
      "unmountOnExit": false,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appear": false,
        "children": [Function],
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "timeout": 500,
        "unmountOnExit": false,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
